domain/consensus/datastructures/pruningstore/imported_pruning_point.go

```diff
@@ -0,0 +1,213 @@
+)
+
+var importedPruningPointUTXOsBucket = database.MakeBucket([]byte("imported-pruning-point-utxos"))
+var importedPruningPointMultiset = database.MakeBucket(nil).Key([]byte("imported-pruning-point-multiset"))
+
+func (ps *pruningStore) ClearImportedPruningPointUTXOs(dbContext model.DBWriter) error {
+	cursor, err := dbContext.Cursor(importedPruningPointUTXOsBucket)
+	if err != nil {
+		return err
+	}
+
+	for cursor.Next() {
+		key, err := cursor.Key()
+		if err != nil {
+			return err
+		}
+		err = dbContext.Delete(key)
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (ps *pruningStore) AppendImportedPruningPointUTXOs(dbTx model.DBTransaction,
+	outpointAndUTXOEntryPairs []*externalapi.OutpointAndUTXOEntryPair) error {
+
+	for _, outpointAndUTXOEntryPair := range outpointAndUTXOEntryPairs {
+		key, err := ps.importedPruningPointUTXOKey(outpointAndUTXOEntryPair.Outpoint)
+		if err != nil {
+			return err
+		}
+		serializedUTXOEntry, err := ps.serializeUTXOEntry(outpointAndUTXOEntryPair.UTXOEntry)
+		if err != nil {
+			return err
+		}
+		err = dbTx.Put(key, serializedUTXOEntry)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (ps *pruningStore) ImportedPruningPointUTXOIterator(dbContext model.DBReader) (model.ReadOnlyUTXOSetIterator, error) {
+	cursor, err := dbContext.Cursor(importedPruningPointUTXOsBucket)
+	if err != nil {
+		return nil, err
+	}
+	return ps.newCursorUTXOSetIterator(cursor), nil
+}
+
+type utxoSetIterator struct {
+	cursor model.DBCursor
+}
+
+func (ps *pruningStore) newCursorUTXOSetIterator(cursor model.DBCursor) model.ReadOnlyUTXOSetIterator {
+	return &utxoSetIterator{cursor: cursor}
+}
+
+func (u *utxoSetIterator) First() bool {
+	return u.cursor.First()
+}
+
+func (u *utxoSetIterator) Next() bool {
+	return u.cursor.Next()
+}
+
+func (u *utxoSetIterator) Get() (outpoint *externalapi.DomainOutpoint, utxoEntry externalapi.UTXOEntry, err error) {
+	key, err := u.cursor.Key()
+	if err != nil {
+		panic(err)
+	}
+
+	utxoEntryBytes, err := u.cursor.Value()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	outpoint, err = deserializeOutpoint(key.Suffix())
+	if err != nil {
+		return nil, nil, err
+	}
+
+	utxoEntry, err = deserializeUTXOEntry(utxoEntryBytes)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	return outpoint, utxoEntry, nil
+}
+
+func (ps *pruningStore) importedPruningPointUTXOKey(outpoint *externalapi.DomainOutpoint) (model.DBKey, error) {
+	serializedOutpoint, err := ps.serializeOutpoint(outpoint)
+	if err != nil {
+		return nil, err
+	}
+
+	return importedPruningPointUTXOsBucket.Key(serializedOutpoint), nil
+}
+
+func (ps *pruningStore) serializeOutpoint(outpoint *externalapi.DomainOutpoint) ([]byte, error) {
+	return proto.Marshal(serialization.DomainOutpointToDbOutpoint(outpoint))
+}
+
+func (ps *pruningStore) serializeUTXOEntry(entry externalapi.UTXOEntry) ([]byte, error) {
+	return proto.Marshal(serialization.UTXOEntryToDBUTXOEntry(entry))
+}
```

Comment:
nit, make these also free functions (it's a little weird that deserialize are free but serialize are not)

Reply:
(Or just move /domain/consensus/datastructures/consensusstatestore/utxo_serialization.go to somewhere public)